ports/esp32/modnetwork.c

```diff
@@ -398,7 +398,26 @@ STATIC mp_obj_t esp_status(size_t n_args, const mp_obj_t *args) {
             }
             return list;
         }
+        case (uintptr_t)MP_OBJ_NEW_QSTR(MP_QSTR_rssi): {
+            ESP_LOGD("wifi-status", "RSSI call");
+            // return signal of AP, only in STA mode
+            require_if(args[0], WIFI_IF_STA);
 
+            wifi_mode_t mode;
+            ESP_EXCEPTIONS(esp_wifi_get_mode(&mode));
+            if ((mode & WIFI_MODE_STA) == 0) {
+                nlr_raise(mp_obj_new_exception_msg(&mp_type_OSError, "STA must be active"));
+            }
+
+            wifi_ap_record_t info;
+            if(!esp_wifi_sta_get_ap_info(&info)) {
```

Comment:
The result of this call can be wrapped in `ESP_EXCEPTIONS()`, which will simplify this part (no need to return `mp_const_none` in the fail path).

Reply:
Fell free to enhance it, I just made working prototype. Enough for our case to show signal bar. More heads, more ideas. :)